API/Services/BookService.cs

```diff
@@ -425,10 +426,189 @@ private async Task InlineStyles(HtmlDocument doc, EpubBookRef book, string apiBa
         }
     }
 
-    public ComicInfo? GetComicInfo(string filePath)
+    private int FindInBuffer(byte[] buffer, int bufLen, byte[] match)
+    {
+        int maxPos = bufLen - match.Length;
+        for (int pos = 0; pos<=maxPos; ++pos)
+        {
+            bool found = true;
+            for (int ch = 0; ch<match.Length; ++ch)
+            {
+                if (buffer[pos+ch] != match[ch])
+                {
+                    found = false;
+                    break;
+                }
+            }
+            if (found)
+            {
+                return pos;
+            }
+        }
+        return -1;
+    }
+
+    private string? GetTextFromXmlNode(XmlDocument doc, XmlNamespaceManager ns, string path)
```

Comment:
I would move all this PDF stuff into a PDF helper class. Because there are so many methods that it's polluting the book service.